smriprep/utils/misc.py

```diff
@@ -21,3 +21,53 @@ def apply_lut(in_dseg, lut, newpath=None):
                    segm.affine, hdr).to_filename(out_file)
 
     return out_file
+
+
+def fs_isRunning(subjects_dir, subject_id, mtime_tol=86400):
+    """
+    Checks FreeSurfer subjects dir for presence of recon-all blocking ``IsRunning`` files,
+    and optionally removes any based on the modification time.
+
+    Parameters
+    ----------
+    subjects_dir : os.PathLike or None
+        Existing FreeSurfer subjects directory
+    subject_id : str
+        Subject label
+    mtime_tol : int
+        Tolerance time (in seconds) between current time and last modification of ``recon-all.log``
+
+    Returns
+    -------
+    subjects_dir : os.PathLike or None
+
+    """
+    import logging
+    from pathlib import Path
+    import time
+
+    if subjects_dir is None:
+        return subjects_dir
+    subj_dir = Path(subjects_dir) / subject_id
+    if not subj_dir.exists():
+        return subjects_dir
+
+    isrunning = tuple(subj_dir.glob("**/IsRunning*"))
+    if not isrunning:
+        return subjects_dir
+    try:
+        reconlog = next(subj_dir.glob("**/recon-all.log"))
+    except StopIteration:
+        # problems finding recon-all.log, just return subjects_dir
+        return subjects_dir
+
+    # if any IsRunning files are found, first check on last modified time of log
+    if (time.time() - reconlog.stat().st_mtime) < mtime_tol:
+        return subjects_dir
```

Comment:
We know at this point that it will not successfully run. Does it make sense instead to raise a readable exception?

```suggestion
    reconlog = subj_dir / "scripts" / "recon-all.log"
    if not reconlog.is_file() or time.time() - reconlog.stat().st_mtime < mtime_tol:
        raise RuntimeError("{subj_dir} contains IsRunning files: {isrunning}\n"
                           "FreeSurfer will not run if these are present, to avoid "
                           "interfering with a running process. If no process is running, "
                           "they may be safely removed.")
```